token/services/network/common/rws/translator/translator.go

```diff
@@ -327,36 +302,46 @@ func (w *Translator) commitSetupAction(setup SetupAction) error {
 
 func (w *Translator) commitIssueAction(issueAction IssueAction) error {
 	base := w.counter
+	graphNonHiding := !issueAction.IsGraphHiding()
 
+	// store outputs
 	outputs, err := issueAction.GetSerializedOutputs()
 	if err != nil {
 		return err
 	}
 	for i, output := range outputs {
-		outputID, err := keys.CreateTokenKey(w.TxID, base+uint64(i))
+		// store output
+		outputID, err := w.KeyTranslator.CreateOutputKey(w.TxID, base+uint64(i))
 		if err != nil {
 			return errors.Errorf("error creating output ID: %s", err)
 		}
-		if err := w.RWSet.SetState(w.namespace, outputID, output); err != nil {
+		if err := w.RWSet.SetState(outputID, output); err != nil {
 			return err
 		}
+		if graphNonHiding {
+			// store also the serial number of this output.
+			// the serial number is used to check that the token exists at time of spending
+			sn, err := w.KeyTranslator.CreateOutputSNKey(w.TxID, base+uint64(i), output)
+			if err != nil {
+				return errors.Errorf("error creating output ID: %s", err)
+			}
+			if err := w.RWSet.SetState(sn, []byte{1}); err != nil {
```

Comment:
Maybe we make `[]byte{1}`  a constant that we call `NotEmpty`?

Reply:
okay